internal/experiment/generation/generation.go

```diff
@@ -111,3 +123,51 @@ func loadApplicationData(app *redskyappsv1alpha1.Application, src string) ([]byt
 
 	return ioutil.ReadFile(dst)
 }
+
+var metricSelectorPattern = regexp.MustCompile(`([a-zA-Z_][a-zA-Z0-9_]*)(=|!=|=~|!~)"([a-zA-Z0-9\-|]+)"`)
+
+// convertPrometheusSelector converts a Prometheus metric selector to a Kubernetes
+// label selector. This is necessary because objectives like "Requests" define their
+// "selector" as a Prometheus selector: in order for that to work with a metric
+// with `type: kubernetes` (or `type: ""`), we must first convert it over.
+func convertPrometheusSelector(metricSelector string) (*metav1.LabelSelector, error) {
+	if metricSelector == "" {
+		return nil, nil
+	}
+
+	labelSelector := &metav1.LabelSelector{}
+	for _, ms := range strings.Split(metricSelector, ",") {
+		parts := metricSelectorPattern.FindStringSubmatch(ms)
+		if len(parts) != 4 {
```

Comment:
I'm only catching 3 parts for the regex submatch
- `([a-zA-Z_][a-zA-Z0-9_]*)`
- `(=|!=|=~|!~)`
- `([a-zA-Z0-9\-|]+)`

testing with 
```
ms := `key="value"`
```

Reply:
Group 0 is the entire match.